### DISCUSSION TITLE
Should we and how to add a de-dup functionality to rclip?

### DISCUSSION THREAD

**Discussion by yurijmikhalevich** [yurijmikhalevich/rclip#23] [Ideas]
This discussion is based on an idea proposed by @ramayer [here](https://github.com/yurijmikhalevich/rclip/pull/21#discussion_r744177051).

The simplest way will be to add a new mode:

```sh
$ rclip --de-dup
```

or

```sh
$ rclip -d
```

If called without any params, it will output the list of all possible duplicates (grouped). If called with a param `rclip -d path/to/an/image`, it will output all images that are possible duplicates of the referenced image.

As for "should we," we first should first test how well it will work.

Am I missing something? What do you think about this proposal? Do you have any other ideas?

**Comment by TheUnstoppablesAI** [comment]
I just saw that this was from 2021! Yes a dedup feature would be very useful. When I backup my pictures I'll just throw them in the same folder. Using rclip to be able to search and then delete results would be useful.
